plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/provider/TerminalVolumeProvider.java

```diff
@@ -43,8 +46,16 @@
     private String terminalArchivePath;
 
     @Inject
-    @Named("che.docker.volumes_agent_options")
-    private String volumeOptions;
+    @Nullable @Named("che.docker.volumes_agent_options")
+    private String agentVolumeOptions;
+
+    public TerminalVolumeProvider() {
+        if (!Strings.isNullOrEmpty(agentVolumeOptions)) {
```

Comment:
Constructor is called before field injection in Guice. So here we will have null value always. Can you replace field injection with a constructor injection.